charmcraft/utils.py

```diff
@@ -185,3 +186,26 @@ def get_host_architecture():
     """Get host architecture in deb format suitable for base definition."""
     os_platform = get_os_platform()
     return ARCH_TRANSLATIONS.get(os_platform.machine, os_platform.machine)
+
+
+def confirm_with_user(prompt, default=False) -> bool:
```

Comment:
argparse + click might be good thing to look into

Reply:
I know click has `confirm()` but am unsure what you are asking here. Look into using click and replacing the argparse bits?

Reply:
not asking, just a random thought in a PR. Sorry :-)